templates/solochain/runtime/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,119 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{AccountId, BalancesConfig, RuntimeGenesisConfig, SudoConfig};
+use alloc::{format, vec, vec::Vec};
+use serde_json::Value;
+use sp_consensus_aura::sr25519::AuthorityId as AuraId;
+use sp_consensus_grandpa::AuthorityId as GrandpaId;
+use sp_core::{Pair, Public};
+use sp_genesis_builder::{self, PresetId};
+use sp_keyring::AccountKeyring;
+
+// Returns the genesis config presets populated with given parameters.
+fn testnet_genesis(
+	initial_authorities: Vec<(AuraId, GrandpaId)>,
+	endowed_accounts: Vec<AccountId>,
+	root: AccountId,
+) -> Value {
+	let config = RuntimeGenesisConfig {
+		balances: BalancesConfig {
+			balances: endowed_accounts
+				.iter()
+				.cloned()
+				.map(|k| (k, 1u128 << 60))
+				.collect::<Vec<_>>(),
+		},
+		aura: pallet_aura::GenesisConfig {
+			authorities: initial_authorities.iter().map(|x| (x.0.clone())).collect::<Vec<_>>(),
+		},
+		grandpa: pallet_grandpa::GenesisConfig {
+			authorities: initial_authorities.iter().map(|x| (x.1.clone(), 1)).collect::<Vec<_>>(),
+			..Default::default()
+		},
+		sudo: SudoConfig { key: Some(root) },
+		..Default::default()
+	};
+
+	serde_json::to_value(config).expect("Could not build genesis config.")
+}
+
+// Generate a crypto pair from seed.
+fn get_from_seed<TPublic>(seed: &str) -> <TPublic::Pair as Pair>::Public
+where
+	TPublic: Public,
+{
+	TPublic::Pair::from_string(&format!("//{}", seed), None)
+		.expect("static values are valid; qed")
+		.public()
+}
+
+// Generate authority keys to be passed to the genesis config presets.
+fn authority_keys_from_seed(s: &str) -> (AuraId, GrandpaId) {
+	(get_from_seed::<AuraId>(s), get_from_seed::<GrandpaId>(s))
+}
+
+/// Return the development genesis config.
+pub fn development_config_genesis() -> Value {
+	testnet_genesis(
+		vec![authority_keys_from_seed(&AccountKeyring::Alice.to_seed())],
```

Comment:
Unified or not, why do we need these methods at all here? 
This seems to be the same as this right?:
```
vec![Sr25519Keyring::Alice.public().into(), Ed25519Keyring::Alice.public().into()],
```

Reply:
Ah, seems this is exactly the same and a lot simpler, didn't realize `AuraId` and `GrandpaId` can be mapped to the same types from `sp_keyring`. Fixed here: [cc761df](https://github.com/paritytech/polkadot-sdk/pull/5868/commits/cc761df41aa4232846022e611abe15442d9c92df), thanks!

Reply:
lol, actually that was exactly my point here: https://github.com/paritytech/polkadot-sdk/pull/5868#issuecomment-2384056790

Reply:
ups :sweat_smile:, I understood your reference, but still did not realize that the types from `sp_keyring` can replace the functions. The above example convinced me.